enzyme/Enzyme/Enzyme.cpp

```diff
@@ -79,6 +79,263 @@ llvm::cl::opt<bool>
 
 namespace {
 
+static void
+handleCustomGradient(llvm::Module &M, llvm::GlobalVariable &g,
+                     std::vector<GlobalVariable *> &globalsToErase) {
```

Comment:
Can you simplify this and the following methods a bit -- there seems to be a fair amount of common code.

Reply:
done